### PR TITLE
PPTP-1787 Check your answers- Deregistration page

### PR DESCRIPTION
**Pull request by kiranprakashdigital**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] User Acceptance Tests (UAT) were run locally and have passed.
 - [x] No Accessibility errors/warnings reported by Wave
Screenshot_20220329_182358

### CODE REVIEW

**Review by tonymccrae-ee — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/DeregistrationConnector.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  httpClient: HttpClient,
+  appConfig: AppConfig,
+  metrics: Metrics
+)(implicit ec: ExecutionContext) {
+
+  def deregister(pptReference: String, deregistrationDetails: DeregistrationDetails)(implicit
+    hc: HeaderCarrier
+  ): Future[Either[ServiceError, HttpResponse]] = {
+    val timer = metrics.defaultRegistry.timer("ppt.deregister.timer").time()
+    httpClient.PUT[Option[DeregistrationReason], HttpResponse](
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/DeregistrationConnector.scala:41]: Feels a little weird for a connector to expose an `HttpResponse`. As far as I can tell this isn't actually used (except in the tests); can we just return `Future[Either[ServiceError, Unit]]`?

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/deregistration/DeregisterCheckYourAnswersController.scala [hunk 2]
```diff
@@ -18,35 +18,46 @@ package uk.gov.hmrc.plasticpackagingtax.registration.controllers.deregistration
...
+    authenticate.async { implicit request =>
+      val pptReference = request.pptReference.getOrElse(
+        throw new IllegalStateException("Ppt reference number not found")
+      )
+      deregistrationDetailRepository.get().flatMap {
+        deregistrationDetails =>
+          deregistrationConnector.deregister(pptReference, deregistrationDetails).flatMap {
+            case Right(_) =>
+              deregistrationDetailRepository.delete()
+              Future(Redirect(routes.DeregistrationSubmittedController.displayPage()))
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/deregistration/DeregisterCheckYourAnswersController.scala:57]: Should this be `Future.successful(...)` if we want to immediately return a successful future. Doesn't `Future(...)` actually schedule running (of ...) in another thread?

File: test/uk/gov/hmrc/plasticpackagingtax/registration/connectors/DeregistrationConnectorISpec.scala [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    "return success response" when {
+
+      "valid request is sent" in {
+
+        givenPutDeregistration(pptReference, Status.OK, DeregistrationReason.CeasedTrading.toString)
+
+        val res: Either[ServiceError, HttpResponse] =
+          await(connector.deregister(pptReference, deregistrationDetails))
+
+        res.right.get.status mustBe OK
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/connectors/DeregistrationConnectorISpec.scala:55]: I think we should just be checking that a `Right` is returned here.

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
Looks good; just a couple of minor suggestions.

**Pull request merged**